hs_core/views/utils.py

```diff
@@ -489,8 +492,16 @@ def rename_irods_file_or_folder_in_django(resource, src_name, tgt_name):
             # since resource_file is a FileField which cannot be directly renamed,
             # this old ResourceFile object has to be deleted followed by creation of
             # a new ResourceFile with new file associated that replace the old one
+            logical_file = res_file_obj[0].logical_file if res_file_obj[0].has_logical_file \
+                else None
+
             res_file_obj[0].delete()
-            ResourceFile.objects.create(content_object=resource, resource_file=tgt_name)
+            res_file = ResourceFile.objects.create(content_object=resource, resource_file=tgt_name)
+            res_file.mime_type = get_file_mime_type(res_file.resource_file.name)
+            if logical_file is not None:
+                res_file.logical_file_content_object = logical_file
```

Comment:
I question the use of GenericRelation here. This is simpler than that. A logical file does not vary in type. Thus, this is a simple foreign key relationship.  

In general GenericRelation is abused project-wide, for no particularly good reason. Since it is very poorly supported in Django, this insulates us from being able to utilize Django built-ins to good measure.